packages/block-editor/src/hooks/duotone.js

```diff
@@ -245,6 +249,49 @@ function scopeSelector( scope, selector ) {
 	return selectorsScoped.join( ', ' );
 }
 
+function BlockDuotoneStyles( { name, duotoneStyle, id } ) {
+	const duotonePalette = useMultiOriginPresets( {
+		presetSetting: 'color.duotone',
+		defaultSetting: 'color.defaultDuotone',
+	} );
+
+	const element = useContext( BlockList.__unstableElementContext );
+
+	// Portals cannot exist without a container.
+	// Guard against empty Duotone styles.
+	if ( ! element || ! duotoneStyle ) {
+		return null;
+	}
+
+	let colors = duotoneStyle;
+
+	if ( ! Array.isArray( colors ) && colors !== 'unset' ) {
+		colors = getColorsFromDuotonePreset( colors, duotonePalette );
+	}
+
+	const duotoneSupportSelectors = getBlockSupport(
+		name,
+		'color.__experimentalDuotone'
+	);
+
+	// Extra .editor-styles-wrapper specificity is needed in the editor
+	// since we're not using inline styles to apply the filter. We need to
+	// override duotone applied by global styles and theme.json.
+	const selectorsGroup = scopeSelector(
+		`.editor-styles-wrapper .${ id }`,
+		duotoneSupportSelectors
+	);
```

Comment:
I know @ajlende has a follow up which uses standard Theme JSON class methods to handle this.